shell/platform/windows/window_state.h

```diff
@@ -37,13 +38,24 @@ struct FlutterDesktopPluginRegistrar {
   // The plugin messenger handle given to API clients.
   std::unique_ptr<FlutterDesktopMessenger> messenger;
 
+  // The plugin texture registrar handle given to API clients.
+  FlutterDesktopTextureRegistrar* texture_registrar;
+
   // The handle for the view associated with this registrar.
   std::unique_ptr<FlutterDesktopView> view;
 
   // Callback to be called on registrar destruction.
   FlutterDesktopOnRegistrarDestroyed destruction_handler;
 };
 
+// State associated with the texture registrar.
+struct FlutterDesktopTextureRegistrar {
```

Comment:
The use of structs for internal implementations is a legacy of the initial C++ embedding being a single file without much complexity; it shouldn't be perpetuated (and in fact is being slowly eliminated). New objects should be classes, and follow the normal class process (be declared in their own file, not expose ivars, contain relevant logic such as what's currently inlined in OnAcquireExternalTexture and the flutter_windows.cc functions, etc.).